File: ITPMApp（第１回 実演時作成）/app/src/main/java/com/example/itpm_app/activities/EditActivity.java
package com.example.itpm_app.activities;

import android.content.ContentValues;
import android.content.Context;
import android.content.Intent;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;

import com.example.itpm_app.databases.ITPMDataOpenHelper;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;

import androidx.appcompat.app.ActionBar;
import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

import com.example.itpm_app.R;

public class EditActivity extends AppCompatActivity {

    private static final String KEY_ID = "key_id";
    private static final String KEY_TITLE = "key_title";

    private EditText mEditText;
    private int mSelectId = -1;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_edit);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mSelectId = getIntent().getIntExtra(KEY_ID, -1);
        ActionBar actionBar = getSupportActionBar();
        if (actionBar != null) {
            if (mSelectId == -1) {
                // 新規
                actionBar.setTitle(R.string.a_new);
            } else {
                // 編集
                actionBar.setTitle(R.string.edit);
            }
        }
        mEditText = findViewById(R.id.title_edit_text);

        String title = getIntent().getStringExtra(KEY_TITLE);
        if (title != null) {
            mEditText.setText(title);
        }

        FloatingActionButton fab = findViewById(R.id.save_button);
        fab.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {

                String title = mEditText.getText().toString();
                if (title.isEmpty()) {
                    Toast.makeText(EditActivity.this, R.string.error_no_title, Toast.LENGTH_SHORT).show();
                } else {
                    ContentValues contentValues = new ContentValues();
                    contentValues.put(ITPMDataOpenHelper.COLUMN_TITLE, title);
                    SQLiteDatabase db = new ITPMDataOpenHelper(EditActivity.this).getWritableDatabase();
                    if (mSelectId == -1) {
                        db.insert(ITPMDataOpenHelper.TABLE_NAME, null, contentValues);
                    } else {
                        contentValues.put(ITPMDataOpenHelper.COLUMN_ID, mSelectId);
                        db.update(ITPMDataOpenHelper.TABLE_NAME, contentValues, ITPMDataOpenHelper.COLUMN_ID + "=" + mSelectId, null);
                    }
                    db.close();
                    finish();
                }
            }
        });
    }

    public static Intent createIntent(Context context, int id, String title) {
        Intent intent = new Intent(context, EditActivity.class);
        intent.putExtra(KEY_ID, id);
        intent.putExtra(KEY_TITLE, title);
        return intent;
    }

}

File: ITPMApp（第１回 実演時作成）/kanpelibrary/src/main/java/com/example/kanpelibrary/activities/EditActivity.kt
package com.example.kanpelibrary.activities

import android.content.ContentValues
import android.os.Bundle
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import com.example.kanpelibrary.R
import com.example.kanpelibrary.databases.ITPMDataOpenHelper
import com.google.android.material.snackbar.Snackbar
import kotlinx.android.synthetic.main.activity_edit.*

class EditActivity : AppCompatActivity() {

    companion object {
        const val KEY_ID = "key_id"
        const val KEY_TITLE = "key_title"
    }

    private var selectId = -1

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_edit)
        setSupportActionBar(toolbar)

        selectId = intent.getIntExtra(KEY_ID, -1)
        val title = intent.getStringExtra(KEY_TITLE)
        titleEditText.setText(title)

        when (selectId) {
            -1 -> supportActionBar?.title = getString(R.string.a_new)
            else -> supportActionBar?.title = getString(R.string.edit)
        }

        saveButton.setOnClickListener {
            if (titleEditText.text.isEmpty()) {
                // エラー表示
                val inputErrorBar = Snackbar.make(findViewById<View>(R.id.rootContainer), R.string.error_no_title, Snackbar.LENGTH_SHORT)
                inputErrorBar.show()
            } else {
                val contentValues = ContentValues()
                if (selectId != -1) {
                    contentValues.put(ITPMDataOpenHelper._ID, selectId)
                }
                contentValues.put(ITPMDataOpenHelper.COLUMN_TITLE, titleEditText.text.toString())
                val db = ITPMDataOpenHelper(this@EditActivity).writableDatabase
                if (selectId == -1) {
                    // 新規追加
                    db.insert(ITPMDataOpenHelper.TABLE_NAME, null, contentValues)
                } else {
                    // 更新処理
                    db.update(ITPMDataOpenHelper.TABLE_NAME, contentValues, ITPMDataOpenHelper._ID + "=" + selectId, null)
                }
                db.close()
                finish()
            }
        }
    }

}
